lib/spack/spack/solver/asp.py

```diff
@@ -316,8 +316,15 @@ def using_libc_compatibility() -> bool:
     return spack.platforms.host().name == "linux"
 
 
-def c_compiler_runs(compiler: spack.compiler.Compiler) -> bool:
-    return compiler.compiler_verbose_output is not None
+def compiler_runs_for_lang(compiler) -> bool:
+    for lang in ["cc", "cxx", "fc", "f77"]:
+        comp_v_out_for_lang = compiler.try_compiler(lang)
+        if not comp_v_out_for_lang:
+            tty.debug(
+                f"the {lang} compiler {getattr(compiler, lang)} does not exist, or does not run correctly."
+                f" The compiler {compiler.spec} will not be used during concretization."
+            )
+        return comp_v_out_for_lang is not None
```

Comment:
This was not actually finished when you left your comment that paused my work on this, so admittedly, testing was exceptionally minimal, and only on compilers with all four compilers defined (i.e. my local MSVC configuration). 

Why would we revert instead of just _fixing_ the issues? 
I'll fix them, and then iterate from there.